services/offline/src/lib/online-status.ts

```diff
@@ -0,0 +1,52 @@
+import debounce from 'lodash/debounce'
+import { useState, useEffect, useCallback } from 'react'
+
+type milliseconds = number
+interface OnlineStatusOptions {
+    debounceDelay?: milliseconds
+}
+
+interface OnlineStatus {
+    online: boolean
+    offline: boolean
+}
+
+/**
+ * Returns the browser's online status. Updates in response to 'online' and
+ * 'offline' events. By default, debounces updates to once every second to
+ * avoid UI flicker, but that delay can be configured with the
+ * `options.debounceDelay` param.
+ *
+ * TODO: Add option to periodically ping server to check online status.
+ * TODO: Add logic to return a variable indicating unstable connection.
+ *
+ * @param {Object} [options]
+ * @param {Number} [options.debounceDelay] - Timeout delay to debounce updates, in ms
+ * @returns {Object} `{ online, offline }` booleans. Each is the opposite of the other.
+ */
+export function useOnlineStatus(options?: OnlineStatusOptions): OnlineStatus {
+    // initialize state to `navigator.onLine` value
+    const [online, setOnline] = useState(navigator.onLine)
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const updateState = useCallback(
+        debounce(
+            ({ type }: Event) => setOnline(type === 'online'),
+            options?.debounceDelay || 1000
+        ),
+        []
```

Comment:
This needs to have `options` in the dependency array

Reply:
Otherwise when updating the debounce delay based on a prop the callback will never get recreated (should also add a test for this)